pandas/core/indexes/extension.py

```diff
@@ -359,15 +359,13 @@ def insert(self, loc: int, item):
         return type(self)._simple_new(new_arr, name=self.name)
 
     def putmask(self, mask, value):
+        res_values = self._data.copy()
         try:
-            value = self._data._validate_setitem_value(value)
+            res_values.putmask(mask, value)
```

Comment:
`self._data` is defined as `_data: ExtensionArray`

Is ExtensionIndex only for ndarray backed ExtensionArray?  or maybe will need catch AttributeError also eventually?

see also https://github.com/pandas-dev/pandas/pull/35032#issuecomment-695218091 (sort of related).

Reply:
Ah my bad for not checking the code thoroughly before commenting and just looking at the diff.

I wasn't aware that you recently added a NDArrayBackedExtensionIndex.

Doesn't this add more complexity? what is the motivation for this?

Reply:
> Doesn't this add more complexity? what is the motivation for this?

Its just taking methods that are duplicated in CategoricalIndex+DatetimeIndexOpsMixin and putting them in one place.  I guess you could say that "adds complexity" bc there is an additional step in the inheritance chain?